File: src/nim/src/services/domain_services/SampleService.nim
# import json, times, typeinfo, strformat
import json
from strformat import `&`
import times
include ../ApplicationService

type SampleService* = ref object of RootObj

proc fib*(this: SampleService, num: int): JsonNode =
  var results: seq[int]
  
  let start_time = getTime()
  for i in 0..<num:
    results.add(ApplicationService().fib(i))
  let end_time = getTime() - start_time # Duration型
  echo &"{end_time.seconds}.{end_time.microseconds}"

  var data = %*{ # %…配列をJsonNodeにする
    "version": "Nim " & NimVersion,
    "time": &"{end_time.seconds}.{end_time.microseconds}",
    "fib": results
  }
  return data

File: src/python/app/services/domain_services/sample_service.py
# from datetime import datetime
from time import time
import sys
from ..application_service import ApplicationService

class SampleService:
    def fib(self, num):
        start_time = time()
        results = [
            ApplicationService().fib(i)
            for i in range(num)
        ]
        end_time = time() - start_time

        data = {
            'version': sys.version,
            'time': "{0:.4f}".format(end_time),
            'fib': results
        }
        return data
